ko/security.rst

```diff
@@ -18,8 +18,300 @@ CUBRID 보안
+
+.. code-block:: bash
+
+	$ cubrid backupdb -S --separate-keys testdb 
+	Backup Volume Label: Level: 0, Unit: 0, Database testdb, Backup Time: Mon Nov 30 14:34:49 2020
+	$ ls
+	lob  testdb  testdb_bk0_keys  testdb_bk0v000  testdb_bkvinf  testdb_keys
+	testdb_lgar_t  testdb_lgat  testdb_lginf  testdb_vinf
+
+**백업 복구 시 키 파일 선택**
+
+백업 시에 분리된 키 파일은 백업 복구 (restoredb)를 수행할 때 **\\-\\-keys-file-path** 를 통해 복구 시 사용할 키 파일로 지정해 줄 수 있다. 지정한 경로에 올바른 키 파일이 존재하지 않을 경우 백업 복구는 실패한다.
+
+\\-\\-keys-file-path 옵션이 주어지지 않을 경우에는 restoredb에서 다음의 우선순위에 따라 백업 복구 시에 사용할 키 파일을 탐색한다. 올바른 키 파일을 찾을 수 없다면 복구에 실패한다. 
+
+*키 파일 분류*
+
+- 서버 키 파일: 일반적으로 서버 실행 시 참조하는 키 파일로 tde_keys_file_path 시스템 파라미터로 설정되어 있거나 기본 경로에 있는 키 파일.
+- 백업 키 파일: 백업 시에 생성된 키 파일로 백업 볼륨 내에 포함되어 있거나, \\-\\-separate-keys로 분리된 키 파일.
+
+*백업 복구 시 키 파일 탐색 순서*
+
+#.	백업 볼륨이 포함하고 있는 백업 키 파일
+#.	백업 시에 \\-\\-separate-keys로 생성된 백업 키 파일 (e.g. [database_name]_bk0_keys). 이 키 파일은 백업 볼륨과 같은 위치에 존재해야 한다.
+#.	시스템 파라미터 tde-keys-file-path 로 지정된 경로에 있는 서버 키 파일
+#.	데이터 볼륨과 같은 위치에 있는 서버 키 파일 (e.g. [database_name]_keys)
+
+ .. note::
+
+	\(1\)에서 백업 볼륨이 키 파일을 포함하고 있을 경우, 백업 볼륨 압축해제 과정에서 --separate-keys 를 통해 생성한 것과 같은 이름으로 백업 키 파일이 생성된다.
+
+	올바른 키 파일을 찾지 못하더라도 백업 볼륨에 암호화된 데이터가 전혀 없다면 복구에 성공할 수 있다. 하지만, 키 파일이 존재하지 않으므로 이후에 TDE 기능을 사용할 수 없다.
+
+.. note:: **증분 백업의 경우**
+
+    증분 백업되어 여러 백업 볼륨을 사용하여 백업 복구를 수행할 경우, \\-\\-level 로 지정하는 레벨의 백업 키 파일을 사용한다. \\-\\-level 을 지정해주지 않을 경우에는 가장 높은 레벨의 백업 키 파일을 사용한다. 사용하는 키 파일만 존재하면 복구가 가능하며, 해당 키 파일은 반드시 존재해야 한다.
+
+.. note:: **백업 키 파일을 분실한 경우**
+
+    기본적으로 백업 키 파일을 분실한 경우에는 백업 복구를 수행할 수 없다. 하지만 이전 백업 볼륨과 현재 사용하려는 백업 볼륨 사이에 키를 변경하지 않아 백업 시에 데이터베이스에 등록된  이전 백업 볼륨의 백업 키를 \\-\\-keys-file-path로 지정해주면 복구가 가능하다. 마찬가지로 기본 경로의 키 파일에 해당 키에도 백업 시의 키가 존재한다면 백업 복구에 사용할 수 있다.
+
+.. note:: **백업 복구 완료 후 등록 키가 변경되는 경우**
+
+	백업 완료 후 데이터베이스에 등록된 키가 데이터베이스 키 파일에 존재하지 않을 경우, 백업 키 파일이 서버 키 파일로 복사되고 해당 키 파일의 첫 번째 키가 데이터베이스를 암호화하는 마스터 키로 임의로 지정된다. 이는 백업복구 완료 시의 데이터베이스에 등록되어있는 마스터 키가 어떠한 키 파일에도 존재하지 않을 수 있기 때문이다.
+
+.. _tde-algorithm:
+
+암호화 알고리즘
+---------------
+
+큐브리드에서 제공하는 대칭키 암호화 알고리즘은 다음과 같다. 
+
+**TDE 암호화 알고리즘**
+
+=================================  =============  =============
+ 알고리즘                           키 크기        옵션 이름   
+=================================  =============  =============
+ Advanced Encryption Algorithm      256 bits       AES         
+---------------------------------  -------------  -------------
+ ARIA                               256 bits       ARIA        
+=================================  =============  =============
+
+Advanced Encryption Algorithm (AES) 는 미국 표준 기술 연구소 (NIST)에 의해 제정된 암호화 방식으로 세계적으로 널리 사용 되고 있는 알고리즘이다. 안정성이 높고 많은 플랫폼에서 하드웨어 가속 등의 최적화를 지원하여 성능적인 저하가 적다. ARIA (lint to ARIA)는 대한민국의 국가 표준 암호화 알고리즘 중 하나로 경량 환경 및 하드웨어 구현에 최적화된 알고리즘이다.
```

Comment:
typo;
(lint to ARIA)
-> (link to ARIA)

Reply:
```suggestion
Advanced Encryption Algorithm (AES) 는 미국 표준 기술 연구소 (NIST)에 의해 제정된 암호화 방식으로 세계적으로 널리 사용 되고 있는 알고리즘이다. 안정성이 높고 많은 플랫폼에서 하드웨어 가속 등의 최적화를 지원하여 성능적인 저하가 적다. ARIA 는 대한민국의 국가 표준 암호화 알고리즘 중 하나로 경량 환경 및 하드웨어 구현에 최적화된 알고리즘이다.
```